docs/developer/container-goals.md

```diff
@@ -0,0 +1,297 @@
+# Creating Container Goals
+
+An SDM is intended to be a programmable way to deliver your software, no matter how it's architected. In this tutorial, we'll take a look at how you can build your own pipeline for creating [Docker](https://www.docker.com/) images, once new code is pushed to a repository.
+
+Container goals allow you to specify outputs and inputs to connect goal results together. We're going to perform two steps in this tutorial:
+
+1.  We'll build a JAR artifact from a Java project via a Docker image from the [Docker Image Registry](https://www.docker.com/products/image-registry); then
+2.  We'll feed that JAR into [Kaniko](https://github.com/GoogleContainerTools/kaniko) in order to build our application's container image.
+
+## Prerequisite
+
+You should have a general idea of both how the SDM operates, and the steps involved in building a Docker image. You'll also need to make sure that your local terminal has access to a Docker daemon; for example, if you have [minikube](https://github.com/kubernetes/minikube) installed, you can try running:
+
+```
+eval $(minikube docker-env)
```

Comment:
Could be interpreted to require a k8s? This only needs a docker daemon running, correct?